src/main/kotlin/no/nav/cv/eures/util/AvroHandling.kt

```diff
@@ -11,43 +11,36 @@ import org.slf4j.LoggerFactory
 class AvroHandling {
 }
 
-// Julian:
-// This is definitely not the best solution, but unfortunately
-// it's the only one I see for handling both avro versions without
-// ending up in situations where we re-index things and get errors
-// until we finish processing the first part of the topic.
+fun List<Byte>.toHex(): String = joinToString(separator = "") { eachByte -> "%02x".format(eachByte) }
+
 fun ByteArray.toMelding(aktorId: String): Melding {
     val log = LoggerFactory.getLogger(AvroHandling::class.java)
 
-    // NOTE: The newest AVRO version prefixes 6 bytes instead of 4
+    val avroPrefixByteSize = 5 // Byte 0 is magic byte, bytes 1-4 is avro schema version - BEWARE OF VERSION
 
     return try {
-        readDatum(aktorId = aktorId, avroPrefixByteSize = 5)
-    } catch (e: Exception) {
-        log.warn("Failed to decode kafka message for $aktorId in the first try, retrying. Size: $size", e)
-        readDatum(aktorId = aktorId, avroPrefixByteSize = 7)
-    } catch (e: Exception) {
-        log.error("Klarte ikke decode kafka melding for $aktorId. Size: $size", e)
-        throw(e)
-    }
-}
+        if(size < avroPrefixByteSize)
+            throw Exception("Trying to decode a message of only $size bytes, with a prefix of $avroPrefixByteSize")
 
-private fun ByteArray.readDatum(aktorId: String, avroPrefixByteSize: Int): Melding {
-    val log = LoggerFactory.getLogger(AvroHandling::class.java)
+        val compatibleAvroVersion = "Ox00000000"
+        val messageAvroVersion = slice(1..4).toHex()
```

Comment:
Vil ikke messageAvroVersion alltid være konstant nå? Hvilken verdi er det som blir "slicet"?

Reply:
extension function: så det er egt this.slice hvor this er meldingen. Stussa på det samme først

Reply:
byte 1-4 er varsjon av avro skjema